GoogleForCreators/web-stories-wp
# Add initial semantic headings logic

## Context



See #5326 for full context.

We want to automatically change the semantic HTML tag being used for text elements, depending on an element's appearance.

## Summary



This change implements the initial logic for using semantic heading tags, based on font size as the sole criterium.

Large text elements will be wrapped in an H1, H2, H3 or paragraph depending on the font size.

There is only ever one H1 on a single story page.

That means multiple H1 tags within the whole story are possible and OK.

## Relevant Technical Choices



## To-do



## User-facing changes



No visual changes

## Testing Instructions






This PR can be tested by following these steps:

1. Create a new story
2. In the left-hand panel, go to the "Text" tab and insert the text presets (Title 1, Title 2, etc.)
3. Preview the story
4. The HTML tags for the inserted text elements should correspond to their title, i.e. Title 1 should be wrapped in `<h1>`
5. Repeat the same for the _Cover_ text sets, where the title should also be wrapped in `<h1>`
6. Confirm that there is only ever a single `<h1>` element on an individual story page, even after inserting multiple presets or text sets


## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #10394
